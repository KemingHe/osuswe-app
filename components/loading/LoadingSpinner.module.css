.loading-spinner-image {
  animation: spin-image 3s linear infinite;
}

@keyframes spin-image {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
