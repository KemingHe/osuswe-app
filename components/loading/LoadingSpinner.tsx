import Image from 'next/image';
import type { JSX } from 'react';

import styles from '@/components/loading/LoadingSpinner.module.css';

export interface LoadingSpinnerProps {
  message?: string;
}

export function LoadingSpinner({
  message = 'Loading...',
}: LoadingSpinnerProps): JSX.Element {
  return (
    <output className="card text-center" aria-live="polite">
      <div className="card-body items-center gap-3 text-lg">
        <Image
          className={styles['loading-spinner-image']}
          src="/images/osuswe-logo-512.png"
          alt="Loading spinner"
          width={50}
          height={50}
          priority={true}
        />
        <p className="text-center">{message}</p>
      </div>
    </output>
  );
}
