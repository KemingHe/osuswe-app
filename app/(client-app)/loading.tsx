import { LoadingSpinner } from '@/components/loading/LoadingSpinner';
import withCenteredWrapper from '@/hocs/withCenteredWrapper';

export default withCenteredWrapper(LoadingSpinner);
