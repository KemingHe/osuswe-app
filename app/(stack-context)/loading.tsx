import { LoadingSpinner } from '@/components/loading/LoadingSpinner';
import { STACK_PROVIDER_LOADING_MESSAGE } from '@/constants/loadingMessageConstants';
import withCenteredWrapper from '@/hocs/withCenteredWrapper';

export default withCenteredWrapper(() => (
  <LoadingSpinner message={STACK_PROVIDER_LOADING_MESSAGE} />
));
