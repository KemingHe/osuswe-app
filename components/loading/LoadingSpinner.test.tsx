import { type RenderResult, render, screen } from '@testing-library/react';

import { LoadingSpinner } from '@/components/loading/LoadingSpinner';

vi.mock('@/constants/loadingMessageConstants', () => ({
  DEFAULT_LOADING_MESSAGE: 'default loading message',
}));

describe('LoadingSpinner component', () => {
  it('renders an accessible and visible loading spinner', () => {
    render(<LoadingSpinner />);

    // Select the element by its "status" role.
    const elementByStatus = screen.getByRole('status');
    expect(elementByStatus).toBeVisible();

    // Select the element by its text content.
    const elementByText = screen.getByText('default loading message');
    expect(elementByText).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment }: RenderResult = render(<LoadingSpinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
