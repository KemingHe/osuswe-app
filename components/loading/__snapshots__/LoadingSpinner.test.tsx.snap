// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSpinner component > matches the snapshot 1`] = `
<DocumentFragment>
  <output
    aria-live="polite"
    class="card size-48"
  >
    <figure
      class="pt-8"
    >
      <img
        alt="Loading spinner"
        class="_loading-spinner-image_84a6ee"
        data-nimg="1"
        decoding="async"
        height="60"
        src="/_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=128&q=75"
        srcset="/_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fosuswe-logo-512.png&w=128&q=75 2x"
        style="color: transparent;"
        width="60"
      />
    </figure>
    <div
      class="card-body px-6 pt-4 pb-8 text-center text-lg leading-snug"
    >
      <p
        class="h-full flex items-center justify-center"
      >
        Loading...
      </p>
    </div>
  </output>
</DocumentFragment>
`;
